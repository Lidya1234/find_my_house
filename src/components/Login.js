import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useHistory, Link } from 'react-router-dom';
import { loginuser } from '../reducers/findSlice';
import '../style/style.css';
import '../style/main.scss';
import Notify from '../Notification/Notify';

const Login = () => {
  let username = '';
  let password = '';
  const history = useHistory();
  const dispatch = useDispatch();
  const { userInfo } = useSelector((state) => state.houses);
  const handleSubmit = (event) => {
    event.preventDefault();
    let user = {
      username,
      password,
    };
    dispatch(loginuser(user));
    user = {
      username: '',
      password: '',
    };
    if (userInfo.logged_in) {
      history.push('/HouseList');
      Notify();
    }
  };

  const handleUNameChange = (e) => {
    username = e.target.value;
  };
  const handlePasswordChange = (e) => {
    password = e.target.value;
  };
  return (
    <div className="container" data-testid="login">
      <div className="login">
        <h1>Sign in</h1>
        <span>
          Hello there! Signin and
          <div>Start managing your system</div>
        </span>

        <form onSubmit={handleSubmit} className="form">
          <div>
            <input
              type="text"
              placeholder="Enter username"
              className="txt-input"
              name="username"
              onChange={handleUNameChange}
            />
          </div>
          <div>
            <input
              type="password"
              placeholder="Enter password"
              className="txt-input"
              name="password"
              onChange={handlePasswordChange}
            />
          </div>
          <div>
            <button type="submit" className="txt-input btn-signin">
              Sign In
            </button>
          </div>
        </form>
        <div>
          You do not have an account
          <Link to="/Signup" className="signlink">Sign up</Link>
        </div>
      </div>
    </div>
  );
};
export default Login;
