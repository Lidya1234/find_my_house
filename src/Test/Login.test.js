import { render, screen, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../reducers/index';
import Login from '../components/Login';

afterEach(() => {
  cleanup();
});

describe('House component tests', () => {
  render(
    <Provider store={store}>
      <Router>
        <Login />
      </Router>
    </Provider>,
  );
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Login />
        </Router>
      </Provider>,
    )
    .toJSON();

  const loginElement = screen.getByTestId('login');

  test('Component should Render', () => {
    expect(loginElement).toBeInTheDocument();
  });
  test('Component should contain a sign in text', () => {
    expect(loginElement).toHaveTextContent('Sign in');
  });
  test('Matches snapshot', () => {
    expect(tree).toMatchSnapshot();
  });
});
