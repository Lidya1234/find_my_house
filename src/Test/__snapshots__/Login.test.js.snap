// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`House component tests Matches snapshot 1`] = `
<div
  className="container"
  data-testid="login"
>
  <div
    className="login"
  >
    <h1>
      Sign in
    </h1>
    <span>
      Hello there! Signin and
      <div>
        Start managing your system
      </div>
    </span>
    <form
      className="form"
      onSubmit={[Function]}
    >
      <div>
        <input
          className="txt-input"
          name="username"
          onChange={[Function]}
          placeholder="Enter username"
          type="text"
        />
      </div>
      <div>
        <input
          className="txt-input"
          name="password"
          onChange={[Function]}
          placeholder="Enter password"
          type="password"
        />
      </div>
      <div>
        <button
          className="txt-input btn-signin"
          type="submit"
        >
          Sign In
        </button>
      </div>
    </form>
    <div>
      You do not have an account
      <a
        className="signlink"
        href="/Signup"
        onClick={[Function]}
      >
        Sign up
      </a>
    </div>
  </div>
</div>
`;
